cmd/metal-api/internal/metal/provisioning.go

```diff
@@ -4,9 +4,56 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/looplab/fsm"
 	"go.uber.org/zap"
 )
 
+// FSM states
+const (
+	FsmStateInitial          string = "Initial State"
+	FsmStatePXEBooting       string = "PXE Booting state"
+	FsmStatePreparing        string = "Rreparing state"
+	FsmStateRegistering      string = "Registering state"
+	FsmStateWaiting          string = "Waiting state"
+	FsmStateInstalling       string = "Installing state"
+	FsmStateBootingNewKernel string = "Booting New Kernel state"
+	FsmStateProvisioned      string = "Provisioned state"
+	FsmStatePlannedReboot    string = "Planned Reboot state"
+	FsmStateCrashed          string = "Crashed state"
+	FsmStateAlive            string = "Alive state"
+	FsmStateResestFailCount  string = "Reset Fail Count state"
+)
+
+var allStates = []string{
+	FsmStateInitial,
+	FsmStatePXEBooting,
+	FsmStatePreparing,
+	FsmStateRegistering,
+	FsmStateWaiting,
+	FsmStateInstalling,
+	FsmStateBootingNewKernel,
+	FsmStateProvisioned,
+	FsmStatePlannedReboot,
+	FsmStateCrashed,
+	FsmStateAlive,
+	FsmStateResestFailCount,
+}
+
+var events = []fsm.EventDesc{
+	{Name: string(ProvisioningEventPXEBooting), Src: []string{FsmStateInitial, FsmStatePlannedReboot, FsmStatePXEBooting, FsmStateCrashed, FsmStateProvisioned, FsmStateResestFailCount}, Dst: FsmStatePXEBooting},
```

Comment:
To improve readability I want to propose some line breaks here.

This way, we would also be able to comment some transitions that are not directly obvious, e.g. `// PXE Boot events practically occur multiple times in a row, which is considered fine`